app/models/requests.go

```diff
@@ -205,12 +204,18 @@ func (r *SendMessageRequest) SetAttributesFromForm(values url.Values) {
 		}
 
 		stringValue := values.Get(fmt.Sprintf("MessageAttribute.%d.Value.StringValue", i))
-		binaryValue := values.Get(fmt.Sprintf("MessageAttribute.%d.Value.BinaryValue", i))
+		encodedBinaryValue := values.Get(fmt.Sprintf("MessageAttribute.%d.Value.BinaryValue", i))
+
+		binaryValue, err := base64.StdEncoding.DecodeString(encodedBinaryValue)
+		if err != nil {
+			log.Warnf("Failed to base64 decode. %s may not have been base64 encoded.", encodedBinaryValue)
+			continue
+		}
 
 		r.MessageAttributes[name] = MessageAttribute{
 			DataType:    dataType,
 			StringValue: stringValue,
-			BinaryValue: []byte(binaryValue),
+			BinaryValue: binaryValue,
```

Comment:
This fixes Bug 1 "Received Message Attribute via XML is not decoded from base 64".
JSON requests has decoded from base 64 on JSON marshaller, but for XML, here `SetAttributesFromForm` does not have decoding step. So internal `BinaryValue` was still base 64 string (as binary) value.

Reply:
I don't really love this.  I know it's more true to life, since that input is supposed to be encoded, but I'd prefer to make the decoding optional.  It's a testing tool after all, and we're doing our best to cover the needs of all.  It's not like we're throwing errors for invalid attributes or anything.

So maybe instead of your continue, the log is fine, but take the value as is and stuff it in the resulting attribute?

Reply:
OK! I will update it as the below 👍 
* Try decoding
    * Success: store it as the decoded raw value
    * Failure: log and store it as original requested value

Reply:
Sorry @kojisaikiAtSony - I'm looking at this again, and more of my previous thoughts are coming back to me.  Looks to me like this change is decoding it, storing it internally, and then upon receiving the message, encoding it, and returning it.  Double check me, is that what you're doing here?

If so, why do this at all?  That's why I undid all this decoding/encoding business not long ago.  It didn't seem like there was a point.  Are you having trouble with the checksum or something because it's encoded?   Where are you seeing the "Received Message Attribute via XML is not decoded from base 64" error, and how are you recreating it?